crates/composer/src/api.rs

```diff
@@ -0,0 +1,29 @@
+use eyre::Result;
+use serde::{Deserialize, Serialize};
+use std::net::UdpSocket;
+
+pub const DEFAULT_SERVER_ADDRESS: &str = "localhost:8888";
+
+#[derive(Serialize, Deserialize, Debug)]
+pub enum Event {
+    TestTick,
+}
+
+pub struct Client {
+    socket: UdpSocket,
+}
+
+impl Client {
+    pub fn new(server_address: Option<&str>) -> Result<Self> {
+        let socket = UdpSocket::bind("localhost:0")?;
```

Comment:
A newbie question: What does binding to port 0 do in this case?

Reply:
@skywhale `0` asks the system to assign some unoccupied port. That's fine here, as the client is binding to this port just to *send* data out and we don't care what particular number that port is.